sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaSourceConsumerFn.java

```diff
@@ -217,26 +239,44 @@ private static Instant ensureTimestampWithinBounds(Instant timestamp) {
   public ProcessContinuation process(
       @Element Map<String, String> element,
       RestrictionTracker<OffsetHolder, Map<String, Object>> tracker,
-      OutputReceiver<T> receiver)
-      throws Exception {
-    Map<String, String> configuration = new HashMap<>(element);
-
+      OutputReceiver<T> receiver) {
     // Adding the current restriction to the class object to be found by the database history
     register(tracker);
+    Map<String, String> configuration = new HashMap<>(element);
```

Comment:
If the common case is that the configuration is only needed for creating a new instance move the new hashmap call to when we actually need to create a copy and instead rely on generating the cache key from `element`

Reply:
done thanks